### DISCUSSION TITLE
Wait for busybox syslogd to be ready

### DISCUSSION THREAD

**Discussion by markand** [davmac314/dinit#313] [General]
Hi there,

I'm trying dinit on alpine to ditch openrc for this amazing init system. I'm currently having fun with it and I like it very much. I'm still tinkering about process management and more especially readiness.

I have syslogd starting early and then crond, klogd, ntpd and so on. But the system is so fast that crond and ntpd are already ready before syslogd actually created /var/log/messages and the catch of /dev/log meaning that a few lines are mixed in the boot process.

```
[  OK  ] multi.target
Apr  7 13:36:22 (none) syslog.info syslogd started: BusyBox v1.36.1
Apr  7 11:36:22 (none) kern.info kernel: [    0.000000] psci: Using standard PSCI v0.2 function IDs
Apr  7 11:36:22 (none) kern.info kernel: [    0.000000] psci: Trusted OS migration not required
Apr  7 11:36:22 (none) kern.info kernel: [    0.000000] psci: SMC Calling Convention v1.0
Apr  7 11:36:22 (none) kern.info kernel: [    0.000000] percpu: Embedded 31 pages/cpu s88552 r8192 d30232 u126976
[  OK  ] mount
[  OK  ] pseudofs
```

The problem is that busybox's syslogd has no way to indicate it is ready so I have no clue how can I improve this, do you have any idea?

Current services files:

```
# cat /etc/dinit.d/syslogd 
type = process
command = /sbin/syslogd -n
restart = true
options = starts-log

# cat /etc/dinit.d/crond 
type = process
command = /usr/sbin/crond -c /etc/crontabs -f
after = syslogd
```

**Comment by davmac314** [comment]
From `man dinit-service(5)`:

              process service.  When starting a bgprocess service, Dinit will
              not consider the service to be fully started until the original
              process forks and terminates.

I.e. `bgprocess`-style services implicitly indicate that they are ready by forking. It's not ideal but it is usable if the process doesn't support any other readiness notification.

Try:
```
type = bgprocess
command = /sbin/syslogd -n
pid-file = /var/run/syslogd.pid
restart = true
options = starts-log
```

A quick web search indicates that `/var/run/syslogd.pid` is the correct location for the pid file, but YMMV and it may not be created depending on how busybox was built.
